src/dbus/server/dbus_thread_object_rcp.cpp

```diff
@@ -2009,22 +2009,33 @@ void DBusThreadObjectRcp::DeactivateEphemeralKeyModeHandler(DBusRequest &aReques
 {
     otError error        = OT_ERROR_NONE;
     auto    threadHelper = mHost.GetThreadHelper();
+    bool    retain_active_session;
+    auto    args = std::tie(retain_active_session);
 
     VerifyOrExit(mBorderAgent.GetEphemeralKeyEnabled(), error = OT_ERROR_NOT_CAPABLE);
 
-    switch (otBorderAgentGetState(threadHelper->GetInstance()))
+    if (DBusMessageToTuple(*aRequest.GetMessage(), args) == OTBR_ERROR_NONE)
     {
-    case OT_BORDER_AGENT_STATE_STOPPED:
-        error = OT_ERROR_FAILED;
-        break;
-    case OT_BORDER_AGENT_STATE_ACTIVE:
-        error = OT_ERROR_INVALID_STATE;
-        break;
-    case OT_BORDER_AGENT_STATE_STARTED:
-        otBorderAgentClearEphemeralKey(threadHelper->GetInstance());
-        break;
+        if (retain_active_session)
+        {
+            switch (otBorderAgentGetState(threadHelper->GetInstance()))
+            {
+            case OT_BORDER_AGENT_STATE_STOPPED:
+                error = OT_ERROR_FAILED;
+                break;
+            case OT_BORDER_AGENT_STATE_ACTIVE:
+                error = OT_ERROR_INVALID_STATE;
+                break;
+            case OT_BORDER_AGENT_STATE_STARTED:
+                otBorderAgentClearEphemeralKey(threadHelper->GetInstance());
+                break;
+            }
+            ExitNow();
+        }
     }
 
+    otBorderAgentDisconnect(threadHelper->GetInstance());
```

Comment:
I think `otBorderAgentClearEphemeralKey` also needs to be called here, if there is no secure session, calling `otBorderAgentDisconnect` wouldn't clear the ephemeral key.

Reply:
Thanks! Modified according to BA state now.